modules/library/jdbc/src/main/java/org/geotools/jdbc/MetadataTablePrimaryKeyFinder.java

```diff
@@ -126,6 +126,8 @@ public PrimaryKey getPrimaryKey(JDBCDataStore store, String schema, String table
                                 String tblName = tablesRs.getString(3);
                                 if (tblName.toLowerCase().equals(tableName.toLowerCase())) {
                                     metadataTableExists = true;
+                                    // some databases are case-sensitive (SingleStore), update table name accordingly
+                                    tableName = tblName;
```

Comment:
will this impact other database implementations?

Reply:
No, all databases by SingleStore have case insensitive table names (as per SQL standard), this change is to support singlestore odd case sensitive default behavior.